openedx/adg/lms/applications/models.py

```diff
@@ -133,15 +137,48 @@ class UserApplication(TimeStampedModel):
     status = models.CharField(
         verbose_name=_('Application Status'), choices=STATUS_CHOICES, max_length=8, default=OPEN,
     )
-    reviewed_by = models.ForeignKey(User, null=True, on_delete=models.CASCADE, verbose_name=_('Reviewed By'), )
+    reviewed_by = models.ForeignKey(
+        User, null=True, blank=True, on_delete=models.CASCADE, verbose_name=_('Reviewed By')
+    )
+    internal_admin_note = models.TextField(null=True, blank=True, verbose_name=_('Admin Note'))
 
     class Meta:
         app_label = 'applications'
-        verbose_name = _('User Application')
-        verbose_name_plural = _('User Applications')
+        verbose_name = _('Application')
+        ordering = ['created']
 
     def __str__(self):
-        return 'UserApplication {id}, for user {email}'.format(id=self.id, email=self.user.email)
+        return '{}'.format(self.user.get_full_name())
+
+    @property
+    def is_cover_letter_provided(self):
+        return self.cover_letter_text or self.cover_letter_file
+
+    @property
+    def prereq_course_scores(self):
+        """
+        Fetch and return applicant scores in the pre-requisite courses of the franchise program
+
+        Returns:
+            list: List of prereq course name and score pairs
+        """
+        prereq_course_overviews = list(
+            CourseOverview.objects.filter(
+                id__in=CourseMeta.objects.filter(is_prereq=True).values_list('course', flat=True)
+            )
+        )
```

Comment:
Let's please move it inside the model, We would have to add a new manager. Also we don't need to convert it into a list because if are just iterating over it. We can iterate over the query set